Commands/Select.java

```diff
@@ -1,55 +1,86 @@
 package Commands;
 
-public class Select extends BaseCommand {
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
-    char identifier = 's';
-    String syntax = identifier + " ([unit]|[x]) [y]";
-    String description = "Select: Selects and highlights a unit based on ID or square coordinates.";
-    byte minArguments = 0;
-    byte maxArguments = 2;
-    String[] arguments;
+public class Select extends Command {
 
-    public Select(String[] args) {
-        arguments = args;
+    private final static byte maxArguments = 2;
+    private final static List<String> arguments = new ArrayList<>();
+    private final static Map<Integer, String> usages = new HashMap<>();
+
+    public Select() {
+        usages.put(0, "");
+        usages.put(1, "(unit)");
+        usages.put(2, "(x) (y)");
+        CommandList.registerAlias("select", this);
+        CommandList.registerAlias("sel", this);
+        CommandList.registerAlias("s", this);
     }
 
     @Override
-    public char getIdentifier() {
-        return identifier;
+    public void setArguments(List<String> args) {
+        arguments.clear();
+        arguments.addAll(args);
     }
 
     @Override
-    public String getSyntax() {
-        return "Usage: " + syntax;
+    public List<String> getArguments() {
+        return arguments;
     }
 
     @Override
-    public String getDescription() {
-        return description;
+    public Map<Integer, String> getUsages() {
+        return usages;
     }
 
     @Override
-    public String getError_tooManyArguments() {
-        return tooManyArguments;
+    public String getName() {
+        return "Select";
     }
 
     @Override
-    public String getError_tooFewArguments() {
-        return tooFewArguments;
+    public String getDefaultAlias() {
+        return "select";
     }
 
     @Override
-    public boolean hasEnoughArguments(String[] arguments) {
-        return arguments.length >= minArguments;
+    public String getBasicUsage() {
+        return "select [unit]";
     }
 
     @Override
-    public boolean hasTooManyArguments(String[] arguments) {
-        return arguments.length > maxArguments;
+    public String getDescription() {
+        return "Asks a (selected) unit to attack another.";
```

Comment:
Is this supposed to be the same as attack command?

Reply:
No... I think the select command was masquerading as another command to get more attention...